dash/dash.py

```diff
@@ -128,6 +131,14 @@ def _handle_error(error):
         # static files from the packages
         self.css = Css()
         self.scripts = Scripts()
+
+        for method, resource in itertools.chain(
+                ((self.scripts.append_script, x)
+                 for x in (external_script_urls or [])),
+                ((self.css.append_css, x)
+                 for x in (external_css_urls or []))):
+            method({'external_url': resource, 'external_only': True})
+
```

Comment:
I think that we should just add these URLs no matter what. In other words, reserve the `serve_locally` argument _just_ for the component CSS where there will always be a local and a remote version. Slowly, I'd like to move ourselves away from the `app.scripts.method` interface as well.

Users may want to serve the assets locally because `unpkg` is unreliable for them or its blocked for them but load the JS or CSS urls from some other CDN. Or, they might have an internal CDN where they host assets but they can't load external assets from unpkg.

So instead, let's keep this as a private variable or something and then just embed it directly in the `_generate_scirpts_html` function: https://github.com/plotly/dash/blob/54ca4a376ec349639da5140f1e2c42a739815626/dash/dash.py#L312-L332 (or however else you want to organize it) rather than embedding it through `append_script`

Reply:
Ah, nevermind I see that `external_only` will embed it no matter what. OK, then that weakens my argument for not using `append_script`. I still don't really like the `app.scripts...` and `app.css...` namespace and I'd like for us to move away from it. However, I guess I don't have strong feelings about whether we should do this incrementally (i.e. not use it in this PR) or remove it all at once in a major version upgrade down the line.